ios/Exponent/Kernel/ReactAppManager/EXReactAppManager.mm

```diff
@@ -674,11 +681,18 @@ - (NSDictionary * _Nullable)initialPropertiesForRootView
   NSMutableDictionary *props = [NSMutableDictionary dictionary];
   NSMutableDictionary *expProps = [NSMutableDictionary dictionary];
 
-  NSDictionary *errorRecoveryProps = [[EXKernel sharedInstance].serviceRegistry.errorRecoveryManager developerInfoForExperienceId:_appRecord.experienceId];
-  if ([[EXKernel sharedInstance].serviceRegistry.errorRecoveryManager experienceIdIsRecoveringFromError:_appRecord.experienceId]) {
-    [[EXKernel sharedInstance].serviceRegistry.errorRecoveryManager increaseAutoReloadBuffer];
-    if (errorRecoveryProps) {
-      expProps[@"errorRecovery"] = errorRecoveryProps;
+  if (_appRecord.experienceId) {
```

Comment:
Instead of adding this guard, could we instead assert that the experienceId is nonnull? What makes me think this might be correct is that even if an app does a reload on the initial render, we already have the initial props for the root view. The one case I'm not sure about is if an app tries to reload when the update is first evaluated, before the first render:

```ts
import * as Updates from 'expo-updates';
export default Root() { ... render ... }
Updates.reloadAsync();
```

Reply:
Yep, this makes sense, and is doubly safe because these particular methods can handle a nil experienceId anyway. Just for kicks I tested this case, and did indeed get an infinite reload loop as expected (rather than a crash).